sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -543,6 +716,80 @@ graph LR
     Modifiable -->|Finalize| Executable
 ....
 
+==== Executable Graph Update
+
+A graph in the executable state can have the configuration of its nodes modified
+using a concept called graph _update_. This avoids a user having to rebuild and
+finalize a new executable graph when only the inputs & outputs to a graph
+change between submissions. Updates to a node take effect from the next
+submission of a graph and will not affect any previous submissions or in-flight
+executions of the same graph.
+
+The only type of node that is currently supported for updating in a graph is
+kernel execution nodes.
+
+The aspects of a kernel execution node that can be configured during update are:
+
+* Parameters to the kernel.
+* ND-Range of the kernel.
+
+Two methods are provided by the API to the user for performing this update.
+An API for updating the complete graph object, which is most useful when the
+graph was created from a library, and an individual node update API.
+
+To update an executable graph, the `property::graph::updatable_graph` property
+must have been set when the graph was created during finalization. Otherwise, an
+exception will be thrown if a user tries to update an executable graph. This
+guarantee allows the backend to provide a more optimized implementation, if
+possible.
+
+The `property::graph::updatable_graph` property also allows an executable graph
```

Comment:
> For in order queue, the contiguous submission is always allowed.

That's actually not the current spec wording, the spec says "Throws synchronously with error code invalid if a previous submission of graph has yet to complete execution." So even if you use an in-order queue, or set execution dependencies in an out-of-order queue, it will still be an exception.  This is very restrictive, which is why we have https://github.com/reble/llvm/pull/345 open. I imagine that https://github.com/reble/llvm/pull/345 will merge first, and the wording here will updated to reflect that (right now it's a bit of a placeholder)

Reply:
yes, you are correct for the current status.  My option should be more precise: For in order queue, the contiguous submission is expected to always be allowed once the limitation of “host sync for multiple command-buffer submissions” is removed.  And hope https://github.com/reble/llvm/pull/345 to be merged soon, this limitation is really so restrictive.